packages/cdn-location/src/fetchAirportCodeFromCdn.ts

```diff
@@ -0,0 +1,98 @@
+import { Logger, withTimeout } from '@streamr/utils'
+
+const logger = new Logger(module)
+
+const fetchHeader: (url: string, header: string, timeout: number) =>
+    Promise<string | null> = async (url: string, header: string, timeout: number) => {
+        const response = await withTimeout(fetch(url, {
+            method: 'HEAD'
+        }), timeout)
+        return response.headers.get(header)
+    }
+
+export const fetchAirportCodeFromAmazon: (timeout: number) => Promise<string> =
+    async (timeout: number) => {
+
+        const header = await fetchHeader('https://aws.amazon.com', 'X-Amz-Cf-Pop', timeout)
+
+        // parse airport code from the first 3 characters of X-Amz-Cf-Pop header
+
+        if (!header || header.length < 3) {
+            throw new Error('Could not get airport code from Amazon')
+        }
+
+        const airportCode = header.substring(0, 3)
+
+        if (!airportCode) {
```

Comment:
Is this check needed? The previous check
```
if (!header || header.length < 3) {
     throw new Error('Could not get airport code from Amazon')
}
```
should ensure that the airportCode cannot be an empty string.

Reply:
Same comment for `fetchAirportCodeFromFastly` and `fetchAirportCodeFromCloudflare`